bigquery/main.go

```diff
@@ -134,23 +167,104 @@ func (n *bqNotifier) SendNotification(ctx context.Context, build *cbpb.Build) er
 		log.Warningf("Build passes filter but does not have trigger ID: %v, status: %v", n.filter, build.GetStatus())
 	}
 	log.Infof("sending Big Query write for build %q (status: %q)", build.Id, build.Status)
+	if build.ProjectId == "" {
+		return fmt.Errorf("Build missing project id")
+	}
+	if build.Id == "" {
+		return fmt.Errorf("Build missing id")
+	}
+	newRow := bqRow{}
+	newRow.ProjectID = build.ProjectId
+	newRow.ID = build.Id
+	newRow.BuildTriggerID = build.BuildTriggerId
+	newRow.Status = build.Status.String()
+	if newRow.Status == "STATUS_UNKNOWN" {
+		return fmt.Errorf("Build %v has unknown status", newRow.ID)
+	}
+	terminalBuild := false
+	for _, status := range terminalStatusCodes {
+		if status == newRow.Status {
+			terminalBuild = true
+		}
+	}
+	if !terminalBuild {
+		log.Infof("Not writing non-terminal build step %v", newRow.Status)
+		return nil
+	}
+
+	log.Infof("BUILD IMAGES %v", build.GetImages())
+	newRow.Images = []*buildImage{}
+	for _, image := range build.GetImages() {
+		buildImage, err := parseImageManifest(image)
+		if err != nil {
+			return fmt.Errorf("Error parsing image manifest: %v", err)
+		}
+		newRow.Images = append(newRow.Images, buildImage)
+	}
+
+	newRow.Steps = []*buildStep{}
+	createTimePre, err := ptypes.Timestamp(build.CreateTime)
+	if err != nil {
+		return fmt.Errorf("Error parsing createTime: %v", err)
+	}
+	createTime := civil.TimeOf(createTimePre)
+	startTimePre, err := ptypes.Timestamp(build.CreateTime)
+	if err != nil {
+		return fmt.Errorf("Error parsing createTime: %v", err)
+	}
+	startTime := civil.TimeOf(startTimePre)
+	finishTimePre, err := ptypes.Timestamp(build.CreateTime)
+	if err != nil {
+		return fmt.Errorf("Error parsing createTime: %v", err)
+	}
+	finishTime := civil.TimeOf(finishTimePre)
+	newRow.CreateTime = createTime
+	newRow.StartTime = startTime
+	newRow.FinishTime = finishTime
+	newRow.Tags = build.Tags
+	newRow.Env = build.Options.Env
+
+	for _, step := range build.GetSteps() {
+		newStep := &buildStep{}
+		newStep.Name = step.Name
+		newStep.ID = step.Id
+		newStep.Status = step.GetStatus().String()
+		newStep.Args = step.Args
+		startTimePre, err := ptypes.Timestamp(step.Timing.StartTime)
+		if err != nil {
+			return fmt.Errorf("Error parsing start time for step: %v", err)
+		}
+		startTime := civil.TimeOf(startTimePre)
```

Comment:
Suggestion: consider adding a function that does the ptypes.Timestamp+civil.TimeOf conversion all in one shot since you repeat these lines a lot.